PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/CreateProjectUtil.java

```diff
@@ -57,46 +61,61 @@
 
 	private static final String[] Scala_Local_Run_Sample_Data = new String[] {
 			"/hdinsight/templates/scala/scala_local_run/data/sample_movielens_data.txt" };
+	
+	private static final String SourcePomFileRelativePath = "/hdinsight/templates/pom/spark_";
+	
+	private static final String JavaMavenProjectFolder = "/main/java";
+	
+	private static final String ScalaMavenProjectFolder = "/main/scala";
 
 	public static void copyFileTo(@NotNull String[] resources, @NotNull String toPath) {
 		for (int i = 0; i < resources.length; ++i) {
-			InputStream inputStream = CreateProjectUtil.class.getResourceAsStream("/resources" + resources[i]);
-			String toFilePath = StringHelper.concat(toPath, getNameFromPath(resources[i]));
-			try {
-				File toFile = new File(toFilePath);
-
-				FileUtils.copyInputStreamToFile(inputStream, new File(toFilePath));
-
-				// refresh after copy
-				convert(toFile).refreshLocal(IResource.DEPTH_ONE, null);
-			} catch (IOException | CoreException e) {
-				// TODO Auto-generated catch block
-				e.printStackTrace();
-			}
+			String toName = getNameFromPath(resources[i]);
+			
+			copyFileTo(resources[i], toPath, toName);
+		}
+	}
+	
+	public static void copyFileTo(@NotNull String resource, @NotNull String toPath, @NotNull String toName) {
+		InputStream inputStream = CreateProjectUtil.class.getResourceAsStream("/resources" + resource);
+		String toFilePath = StringHelper.concat(toPath, toName);
+		try {
+			File toFile = new File(toFilePath);
+
+			FileUtils.copyInputStreamToFile(inputStream, new File(toFilePath));
+
+			// refresh after copy
+			convert(toFile).refreshLocal(IResource.DEPTH_ONE, null);
+		} catch (IOException | CoreException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
 		}
 	}
 
-	public static void createSampleFile(@NotNull String id, @NotNull String projectName) throws CoreException {
+	public static void createSampleFile(@NotNull String id, @NotNull String projectName, boolean useMaven, SparkVersion sparkVersion) throws CoreException {
 		final IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectName);
 		final IFolder sourceRootFolder = project.getFolder("src");
 
 		if (!sourceRootFolder.exists()) {
 			sourceRootFolder.create(false, true, null);
 		}
-		final String rootPath = sourceRootFolder.getLocation().toFile().getAbsolutePath();
+		String rootPath = sourceRootFolder.getLocation().toFile().getAbsolutePath();
 
 		switch (id) {
 		case "com.microsoft.azure.hdinsight.local-scala.projwizard":
-			createResourceStructForLocalRunScalaProject(sourceRootFolder, rootPath, project);
+			if (useMaven) rootPath += ScalaMavenProjectFolder;
```

Comment:
>if (useMaven) rootPath += ScalaMavenProjectFolder; [](start = 3, length = 50)

format with {}?

Reply:
Fixed.

---
In reply to: [136773369](https://github.com/Microsoft/azure-tools-for-java/pull/984#discussion_r136773369) [](ancestors = 136773369)